kimchi/src/tests/and.rs

```diff
@@ -0,0 +1,352 @@
+fn check_and<G: KimchiCurve>(
+    witness: &[Vec<G::ScalarField>; COLUMNS],
+    bytes: usize,
+    input1: G::ScalarField,
+    input2: G::ScalarField,
+) {
+    let and_row = xor::num_xors(bytes * 8) + 1;
+    let big_in1 = input1.to_biguint();
+    let big_in2 = input2.to_biguint();
+    assert_eq!(witness[3][and_row], input1 + input2);
+    assert_eq!(
+        witness[4][and_row],
+        BigUint::bitxor(&big_in1, &big_in2).into()
+    );
+    assert_eq!(
+        witness[5][and_row],
+        BigUint::bitand(&big_in1, &big_in2, bytes).into()
+    );
+}
+
+fn setup_and<G: KimchiCurve, EFqSponge, EFrSponge>(
+    input1: Option<G::ScalarField>,
+    input2: Option<G::ScalarField>,
+    bytes: usize,
+) -> (
+    ConstraintSystem<G::ScalarField>,
+    [Vec<G::ScalarField>; COLUMNS],
+)
+where
+    G::BaseField: PrimeField,
+    EFqSponge: Clone + FqSponge<G::BaseField, G, G::ScalarField>,
+    EFrSponge: FrSponge<G::ScalarField>,
+{
+    let rng = &mut StdRng::from_seed(RNG_SEED);
+
+    let cs = create_test_constraint_system_and::<G, EFqSponge, EFrSponge>(bytes);
+
+    // Initalize inputs
+    let input1 = rng.gen(input1, Some(bytes * 8));
+    let input2 = rng.gen(input2, Some(bytes * 8));
+
+    let witness = and::create_and_witness(input1, input2, bytes);
+
+    check_and::<G>(&witness, bytes, input1, input2);
+
+    (cs, witness)
+}
+
+fn test_and<G: KimchiCurve, EFqSponge, EFrSponge>(
+    input1: Option<G::ScalarField>,
+    input2: Option<G::ScalarField>,
+    bytes: usize,
+) -> [Vec<G::ScalarField>; COLUMNS]
+where
+    G::BaseField: PrimeField,
+    EFqSponge: Clone + FqSponge<G::BaseField, G, G::ScalarField>,
+    EFrSponge: FrSponge<G::ScalarField>,
+{
+    let (cs, witness) = setup_and::<G, EFqSponge, EFrSponge>(input1, input2, bytes);
+
+    for row in 0..witness[0].len() {
+        assert_eq!(
+            cs.gates[row].verify_witness::<G>(row, &witness, &cs, &witness[0][0..cs.public]),
+            Ok(())
+        );
+    }
+
+    witness
+}
+
+// Function to create a prover and verifier to test the AND circuit
+fn prove_and_verify<G: KimchiCurve, EFqSponge, EFrSponge>(bytes: usize)
+where
+    G::BaseField: PrimeField,
+    EFqSponge: Clone + FqSponge<G::BaseField, G, G::ScalarField>,
+    EFrSponge: FrSponge<G::ScalarField>,
+{
+    let rng = &mut StdRng::from_seed(RNG_SEED);
+
+    // Create
+    let (mut next_row, mut gates) = CircuitGate::<G::ScalarField>::create_and(0, bytes);
+
+    // Temporary workaround for lookup-table/domain-size issue
+    for _ in 0..(1 << 13) {
+        gates.push(CircuitGate::zero(Wire::for_row(next_row)));
+        next_row += 1;
+    }
+
+    // Create inputs
+    let input1 = rng.gen(None, Some(bytes * 8));
+    let input2 = rng.gen(None, Some(bytes * 8));
+
+    // Create witness
+    let witness = and::create_and_witness(input1, input2, bytes);
+
+    TestFramework::<G>::default()
+        .gates(gates)
+        .witness(witness)
+        .lookup_tables(vec![xor::lookup_table()])
```

Comment:
It would be nice if the user didn't need to know that and require xor's lookup.  That is, it would be nice just to always do the same thing: `vec![gate::lookup_table()]`